Source/Plugins/Core/com.equella.core/js/tsrc/mainui/App.tsx

```diff
@@ -0,0 +1,264 @@
+ * limitations under the License.
+ */
+import { ThemeProvider } from "@material-ui/core";
+import { oeqTheme } from "../theme";
+import * as React from "react";
+import { getCurrentUser, UserData } from "../api/currentuser";
+import { ErrorResponse } from "../api/errors";
+import {
+  LegacyContent,
+  LegacyContentProps,
+  PageContent,
+} from "../legacycontent/LegacyContent";
+import { Template, TemplateProps, TemplateUpdate } from "./Template";
+import { defaultNavMessage, NavAwayDialog } from "./PreventNavigation";
+import { shallowEqual } from "shallow-equal-object";
+import { OEQRoute, OEQRouteComponentProps, routes } from "./routes";
+import {
+  Prompt,
+  Redirect,
+  Route,
+  RouteComponentProps,
+  Switch,
+} from "react-router";
+import ErrorPage from "./ErrorPage";
+import { LegacyPage, templatePropsForLegacy } from "./LegacyPage";
+import { BrowserRouter } from "react-router-dom";
+import { LegacyForm } from "../legacycontent/LegacyForm";
+import HtmlParser from "react-html-parser";
+import SettingsPage from "../settings/SettingsPage";
+import { startHeartbeat } from "../util/heartbeat";
+
+const beforeunload = function (e: Event) {
+  e.returnValue = ("Are you sure?" as unknown) as boolean;
+  return "Are you sure?";
+};
+
+const baseFullPath = new URL(document.head.getElementsByTagName("base")[0].href)
+  .pathname;
+export const basePath = baseFullPath.substr(0, baseFullPath.length - 1);
+
+function IndexPage() {
+  const [currentUser, setCurrentUser] = React.useState<UserData>();
+  const [fullPageError, setFullPageError] = React.useState<ErrorResponse>();
+  const errorShowing = React.useRef(false);
+
+  const refreshUser = React.useCallback(() => {
+    getCurrentUser().then(setCurrentUser);
+  }, []);
+
+  React.useEffect(() => refreshUser(), []);
+
+  const [navAwayCallback, setNavAwayCallback] = React.useState<{
+    message: string;
+    cb: (confirm: boolean) => void;
+  }>();
+
+  const [preventNavMessage, setPreventNavMessage] = React.useState<string>();
+  const [legacyContentProps, setLegacyContentProps] = React.useState<
+    LegacyContentProps
+  >({
+    enabled: false,
+    pathname: "",
+    search: "",
+    locationKey: "",
+    userUpdated: refreshUser,
+    redirected: () => {},
+    onError: () => {},
+    render: () => <div />,
+  });
+
+  const [templateProps, setTemplateProps] = React.useState({
+    title: "",
+    fullscreenMode: "YES",
+    children: [],
+  } as TemplateProps);
+
+  const setPreventNavigation = React.useCallback(
+    (prevent) => {
+      const message = prevent ? defaultNavMessage() : undefined;
+      if (message) {
+        window.addEventListener("beforeunload", beforeunload, false);
+      } else {
+        window.removeEventListener("beforeunload", beforeunload, false);
+      }
+      setPreventNavMessage(message);
+    },
+    [setPreventNavMessage]
+  );
+
+  const nonBlankNavMsg = preventNavMessage ? preventNavMessage : "";
```

Comment:
`Msg` is spelled out as `Message` elsewhere in the file, it may be good to make it consistent

```suggestion
  const nonBlankNavMessage = preventNavMessage ? preventNavMessage : "";
```